.ado/publish.yml

```diff
@@ -5,38 +5,156 @@
+      windows:
+        imageName: 'windows-latest'     # Win-specific Py + Plkatform-independent Py
   pool: 
-    vmImage: 'ubuntu-latest'
+    vmImage: $(imageName)
+
   steps:
   - task: UsePythonVersion@0
     inputs:
       versionSpec: '3.11'
 
   - script: |
       python ./prereqs.py --install
+    displayName: Install Prereqs
+
+  - script: |
       npm install -g @vscode/vsce
-    displayName: Install VSCE Pub Prereqs
-    
+    displayName: Install Prereqs for VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+   
   - script: |
       python build.py --wasm --npm --vscode
     displayName: Build VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      python build.py --pip
+    displayName: Build Platform-Dependent Py Packages
       
+  - script: |
+      python build.py --jupyterlab
+    displayName: Build Platform-Independent Py Packages
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
   - script: |
       cp ../LICENSE.txt .
       vsce package $(Ver.Major).$(Ver.Minor).$(Ver.Patch)
     displayName: Pack VSCode Extension
     workingDirectory: '$(System.DefaultWorkingDirectory)/vscode'
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      ls vscode/*
+      ls vscode/out/*
+    displayName: List VSCode Extension
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/vscode/qsharp-vscode-$(Ver.Major).$(Ver.Minor).$(Ver.Patch).vsix
+    artifact: VSIX
+    displayName: Upload VSCode Extension Artifact
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/vscode/out/extension.js
+    artifact: VSIX.EntryPoint
+    displayName: Upload VSCode Ext EntryPoint Artifact
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - script: |
+      dir target\wheels\*
+    displayName: List Py Packages on Win
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
+  - script: |
+      ls target/wheels/*
+    displayName: List Py Packages on non-Win
+    condition: ne(variables['Agent.OS'], 'Windows_NT')
+
+  - publish: $(System.DefaultWorkingDirectory)/target/wheels
+    artifact: Wheels.Mac
+    displayName: Upload Python Artifacts Mac
+    condition: eq(variables['Agent.OS'], 'Darwin')
+
+  - publish: $(System.DefaultWorkingDirectory)/target/wheels
+    artifact: Wheels.Win
+    displayName: Upload Python Artifacts Win
+    condition: eq(variables['Agent.OS'], 'Windows_NT')
+
+
+- job: "Publish"
+  dependsOn: Build
+  pool: 
+    vmImage: 'windows-latest'
+
+  steps:
+  - download: current
+    artifact: Wheels.Win
+    displayName: Download Python Artifacts Win
+
+  - download: current
+    artifact: Wheels.Mac
+    displayName: Download Python Artifacts Mac
+
+  - script: |
+      mkdir target\wheels
+      move ..\Wheels.Win\*.whl target\wheels
+      move ..\Wheels.Mac\*.whl target\wheels
+      dir target\wheels\*
```

Comment:
'dir' doesn't exist in Mac or Linux, so I assume this should be Windows only in a condition?

Reply:
Ah, I see the 'publish' script is Windows only. Never mind.